### DISCUSSION TITLE
Validating JSON encoded values

### DISCUSSION THREAD

**Discussion by khepin** [cuelang/cue#660] [Q&A] [answered]
The JSON I'm working on contains some fields that are themselves encoded JSON. Example:

```json
{
    "field": "{\"key\": \"value\"}"
}
```

How would one validate the json under `field` using cue?

**Comment by mpvl** [comment, marked as answer]
```
import "encoding/json"

field1: json.Valid // is Valid JSON

field2: json.Validate(#Schema). // field must be valid JSON and conform to a given CUE schema

#Schema: { ... }
```

**Reply by khepin** [reply]
tears in my eyes! Very neat!

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/660.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
